crates/consensus/beacon/src/engine/mod.rs

```diff
@@ -660,7 +660,16 @@ where
                     // new VALID update that moved the canonical chain forward
                     let _ = self.update_head(outcome.header().clone());
                 } else {
+                    #[cfg(not(feature = "optimism"))]
                     debug!(target: "consensus::engine", fcu_head_num=?outcome.header().number, current_head_num=?self.blockchain.canonical_tip().number, "Ignoring beacon update to old head");
+
+                    #[cfg(feature = "optimism")]
+                    if self.blockchain.chain_spec().optimism {
+                        warn!(target: "consensus::engine", fcu_head_num=?outcome.header().number, current_head_num=?self.blockchain.canonical_tip().number, "[OPTIMISM] Allowing beacon reorg to old head");
+                        let _ = self.update_head(outcome.header().clone());
+                    } else {
+                        debug!(target: "consensus::engine", fcu_head_num=?outcome.header().number, current_head_num=?self.blockchain.canonical_tip().number, "Ignoring beacon update to old head");
+                    }
```

Comment:
readability nit
```suggestion
                    let fcu_head_num = outcome.header().number;
                    let current_head_num = self.blockchain.canonical_tip().number;

                    #[cfg(not(feature = "optimism"))]
                    debug!(target: "consensus::engine", ?fcu_head_num, ?current_head_num, "Ignoring beacon update to old head");

                    #[cfg(feature = "optimism")]
                    if self.blockchain.chain_spec().optimism {
                        warn!(target: "consensus::engine", ?fcu_head_num, ?current_head_num, "[OPTIMISM] Allowing beacon reorg to old head");
                        let _ = self.update_head(outcome.header().clone());
                    } else {
                        debug!(target: "consensus::engine", ?fcu_head_num, ?current_head_num, "Ignoring beacon update to old head");
                    }
```

Reply:
Outdated - took a similar approach, code around this area changed after rebasing :)